lighthouse-core/fraggle-rock/gather/driver.js

```diff
@@ -83,6 +83,11 @@ class Driver {
     if (!this._session) return;
     await this._session.dispose();
   }
+
+  /** @return {LH.Puppeteer.Page} */
+  getPage() {
```

Comment:
It's not clear to me that we necessarily want to expose this, driver is created from a `Page` but its primary interface is a session, not a page (or target). Maybe it makes sense, but at that point, should `page` just be included in `passContext`? And what is the point of `Driver`?

Reply:
> It's not clear to me that we necessarily want to expose this

This is the best way to support this docs example, I don't think it makes sense for a developer to create their own puppeteer page when we have our own just sitting there. I'm also fine removing this docs example FWIW and just not supporting this pattern.

> Maybe it makes sense, but at that point, should page just be included in passContext?

TBH I don't see much benefit to refactoring the gather context like that. I don't expect this function to be used very much so it's fine leaving it one layer deep on the driver.

> And what is the point of Driver?

It's partially legacy support, because the legacy driver implements the same interface. In the future, we may want to create a separate session for certain tasks, so driver will not have a 1:1 relationship with the session. https://github.com/GoogleChrome/lighthouse/pull/13752.

Reply:
Thinking about this for a while now (sorry), it does still seem like `passContext ` would be the best place for `page`. It looks like it wouldn't require too much plumbing? Adding it to `CollectPhaseArtifactOptions` and then to the `FRTransitionalContext`.

My main concern is we're going to want to start simplifying the world of context/driver/session/target in the new FR-first world, especially when we remove legacy support, and the less stuff layered into driver the better. Long term, having `page` on the gatherer context makes sense to me, but interested in other people's mental models. cc @paulirish if you have thoughts too.

Reply:
I'll try and do it in a separate PR, merge that PR to master, then update this PR so this PR is just a pure docs update.